pkg/authconfig/authconfig_handler.go

```diff
@@ -0,0 +1,36 @@
+package authconfig
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/gin-gonic/gin"
+)
+
+// Service implements the auth config endpoint, which is invoked to 
+// retrieve the auth config for the ui.
+type Service struct {
+	config *configuration.Registry
+	logger *log.Logger
+}
+
+// New returns a new Service instance.
+func New(logger *log.Logger, config *configuration.Registry) *Service {
+	return &Service{
+		logger: logger,
+		config: config,
+	}
+}
+
+// AuthconfigHandler returns signup info.
+func (srv *Service) AuthconfigHandler(ctx *gin.Context) {
+	ctx.Writer.Header().Set("Content-Type", srv.config.GetAuthClientConfigAuthContentType())
+	ctx.Writer.WriteHeader(http.StatusOK)
+	_, err := ctx.Writer.WriteString(srv.config.GetAuthClientConfigAuthRaw())
+	if err != nil {
+		srv.logger.Println("error writing response body", err.Error())
+		http.Error(ctx.Writer, err.Error(), http.StatusInternalServerError)
```

Comment:
As I mentioned before I think we should return payloads in JSON format instead of raw strings (with some exceptions like authconfig payload). Including error responses. So, we will need to introduce `Error` struct which we will marshal to JSON and return in the body. Something like:
```
type Error struct {
    Status string
    Code string
    Title string
    Details string
    ...
}
```
And then construct it when needed: `NewError(http.StatusInternalServerError, "oopsie woopsie... some human readable title", err)`

But let's do it in a separate PR (would need a JIRA for that :) )

Reply:
created a jira for this here: https://jira.coreos.com/browse/CRT-256 :-)